src/Model/HasAttributesTrait.php

```diff
@@ -123,6 +133,12 @@ public function &getDocumentAttribute(string $key)
             return $this->mutableCache[$key];
         }
 
+        if ($caster = CastResolver::resolve($this->casts[$key] ?? null)) {
+            $value = $caster->get($this->attributes[$key] ?? null);
+
+            return $value;
+        }
```

Comment:
As already commented, you could do this:
```suggestion
        if ($cast = $this->casts[$key] ?? null) {
            $caster = CastResolver::resolve($cast);
            $value = $caster->get($this->attributes[$key] ?? null);

            return $value;
        }
```

Reply:
Interesting. I'll give it a try.

Reply:
Done: https://github.com/leroy-merlin-br/mongolid/commit/0fa4607f1c7a1c45c9c09bd8411843350ac6dbe7